Src/libCZI/Doc/mainpage.markdown

```diff
@@ -29,6 +29,15 @@ It is intended to be easily portable to other platforms.
 It aims to be thread-safe (by most [definitions](https://en.wikipedia.org/wiki/Thread_safety)) - which isn't too surprising since it only allows read-only functionality.
 The library itself does not leverage multithreading, but it is designed for being used in a multithreaded environment.
 
+Versioning of libCZI
+====================
+
+For libCZI the [semantic versioning scheme 2.0](https://semver.org/) is to be applied. The version number is defined in the top-level CMakeList.txt file [here](https://github.com/ZEISS/libczi/blob/60be42d839665e3fdc33c14ab5149aadfadd7ddd/CMakeLists.txt#L4).
+This is the only place where the version number is defined. When making changes, this version number needs to be updated accordingly.
```

Comment:
As discussed offline, I am wondering how we can use this to integrate a version bump check into PR validation - i.e. to check for the version being different from what is currently on _main_. Do you have an idea? Use Git Tags?

Reply:
Well... I'd guess we should require that every PR gives an assessment whether it is an "incompatible API change", "added functionality in a backwards compatible manner" or a "backwards compatible bug-fix" (according to the definition [here](https://semver.org/)). This is then to be checked by the reviewer, and it is checked whether the version number is adapted accordingly. Maybe we can add this to the PR-template? Or - those bullets are already making this distinction
![image](https://user-images.githubusercontent.com/4881321/216028740-44ba4ecf-decf-42be-8a33-46ed4a519c22.png)
What would be nice: maybe add a bullet to the checklist -> has the version number been updated accordingly?

Aside from this - now it would be straight-forward to assign a Git-label (giving the version-string), still leaving it a decision to be made "for which version do we assign a label".

Reply:
I will update the PR template for now as suggested. On top - since you talked about labels: I just had the idea to maybe do it the same as with the _cla_ label, i.e. let a project maintainer assign a label that is checked through a GitHub action to be present. The label _version_ then means: It has been checked if a version bump needs to be performed. Makes sense?

Reply:
As discussed - sounds like a good idea to me, my concern here is about KISS and YAGNI.

Reply:
As discussed - only updated the PR Template in Commit [43cba39](https://github.com/ZEISS/libczi/pull/30/commits/43cba3913179405b9b801afd7a6ba2c2a28e7587).